### DISCUSSION TITLE
how to avoid warnings from import static org.assertj.core.api.Assertions.assertThat;

### DISCUSSION THREAD

**Discussion by pmendelson** [assertj/assertj#3588] [General]
Form day 1 i have always created tests with this pattern

import static org.assertj.core.api.Assertions.assertThat;

```java
public class TestClass {
  @Test
  public void testMethod() {
    String tstStr = null;
    assertThat(tstStr).isNull();
  }
}
```

Using the static import allows my tests to look more natural.

Recently some of the assertThat signatures have been deprecated.  Therefore when I look at my code in an IDE like eclipse i get warnings about my static import.

Are their any recommendations to avoid this warnings without having to write

Assertions.assertThat(tstStr).isNull();

**Comment by pmendelson** [comment]
After I wrote this thread I decided to read the javadoc.  I now see 2 replacement classes, [AssertionsForClassTypes](https://javadoc.io/static/org.assertj/assertj-core/3.26.3/org/assertj/core/api/AssertionsForClassTypes.html) and [AssertionsForInterfaceTypes](https://javadoc.io/static/org.assertj/assertj-core/3.26.3/org/assertj/core/api/AssertionsForInterfaceTypes.html).

Thanks for providing such a useful library

**Reply by scordio** [reply]
I couldn't reproduce the issue but I'm using IntelliJ IDEA. Could it be an Eclipse specific issue?

That's what I see in IntelliJ:

![image](https://github.com/user-attachments/assets/eb649cf5-4ffc-49a0-ba87-cf6e275d44f3)

**Reply by pmendelson** [reply]
It could be an eclipse issue but whether it's an eclipse bug or a "normal" reaction to some, but not all, assertThat signatures being deprecated is open to debate

<img width="854" alt="image" src="https://github.com/user-attachments/assets/477e81b0-a5cb-45ba-b336-1dc1866fa6f1">

I have no plans to stop using eclilpse or assertj but i'd love to find a way to eliminate this warning.

**Reply by scordio** [reply]
Has something similar been reported to Eclipse? With a quick search, I could find only groovy/groovy-eclipse#1478.
